fly/fly.py

```diff
@@ -1,32 +1,35 @@
 import json
 import os
+import platform
 import subprocess
 
 import requests
 
 
+DEFAULT_FLY_BIN = '/usr/local/bin/fly'
+
+
 class Fly:
     def __init__(
-            self, concourse_url, executable='/usr/local/bin/fly',
-            platform='darwin', target='default'
+        self,
+        concourse_url,
+        executable=DEFAULT_FLY_BIN,
+        target='default'
     ):
         self.concourse_url = concourse_url
         self.executable = executable
-        self.platform = platform
         self.target = target
 
+    def get_fly(self):
         if not os.path.isfile(self.executable):
-            self._get_fly()
-
-    def _get_fly(self):
-        url = f'{self.concourse_url}/api/v1/cli?arch=amd64&platform=' \
-            f'{self.platform}'
-        response = requests.get(url, stream=True)
-        if response.status_code == 200:
-            with open(self.executable, 'wb') as f:
-                for chunk in response:
-                    f.write(chunk)
-            os.chmod(self.executable, 0o755)
+            url = f'{self.concourse_url}/api/v1/cli?arch=amd64&platform=' \
+                f'{platform.system().lower()}'
+            response = requests.get(url, stream=True)
+            if response.status_code == 200:
+                with open(self.executable, 'wb') as f:
+                    for chunk in response:
+                        f.write(chunk)
+                os.chmod(self.executable, 0o755)
```

Comment:
Although I enjoy the use of octal notation here, perhaps it would be (slightly) more readable if you used the constants, eg:
```python
os.chmod(self.executable, stat.S_IXUSR | stat.S_IXGRP | stat.S_IXOTH | ...)
```

Reply:
Although these constants are horrible too.